### PR TITLE
ref(useOverlay): Update popper position on state change

### PR DESCRIPTION
**Pull request by vuluongj20**
Currently, we do not explicitly tell PopperJS to update the popper element's position on open state change. This update step is [recommended in PopperJS's docs](https://popper.js.org/docs/v2/tutorial/#functionality), and seems to be pretty cheap. 

This is not a problem for us yet — our overlays still render correctly because they are mounted and unmounted on open state change, forcing PopperJS to recompute every time. In the future, however, overlays that don't unmount (whose visibility might be controlled by `display: none;`, for example) will not render correctly.

Here's a sample overlay that has `display: none;` added when it's not open.
**Before:**
Screenshot 2023-01-05 at 10 54 12 AM

**After:**
Screenshot 2023-01-05 at 10 54 34 AM

### CODE REVIEW

**Comment by evanpurkhiser** [comment]
Could you point out what changed exactly here, a bit hard to see since some of the code was moved

File: static/app/utils/useOverlay.tsx [hunk 1]
```diff
@@ -183,10 +158,43 @@ function useOverlay({
...
+
+  // Get props for trigger button
+  const openState = useOverlayTriggerState({isOpen, defaultOpen, onOpenChange});
+  const {buttonProps} = useButton(
+    {
+      onPress: () => {
+        openState.open();
+        popperUpdate?.();
+      },
+    },
```
Comment (vuluongj20) [inline static/app/utils/useOverlay.tsx:175]: @evanpurkhiser Added a call to `popperUpdate()` here. This is the only thing that's changed.

Comment (evanpurkhiser) [reply static/app/utils/useOverlay.tsx:175]: 🆒

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/utils/useOverlay.tsx (hunk 1 above)
Comment (vuluongj20) [reply static/app/utils/useOverlay.tsx]: Actually just moved `popperUpdate()` to `useOverlayTriggerState()` (https://github.com/getsentry/sentry/pull/42824/commits/98afe417d004041282e031374405c6cd60654c88) — same effect, just making sure that the function is fired every time the open state changes and not just when the trigger button is clicked.

**Pull request merged**